aspnetcore/grpc/test-services/sample/Tests/Server/UnitTests/GreeterServiceTests.cs

```diff
@@ -0,0 +1,139 @@
+        #endregion
+
+        [Fact]
+        public async Task SayHelloServerStreamingTest()
+        {
+            // Arrange
+            var mockGreeter = new Mock<IGreeter>();
+            mockGreeter.Setup(m => m.Greet(It.IsAny<string>())).Returns((string s) => $"Hello {s}");
+            var service = new TesterService(mockGreeter.Object);
+
+            var cts = new CancellationTokenSource();
+            var callContext = TestServerCallContext.Create(cancellationToken: cts.Token);
+            var responseStream = new TestServerStreamWriter<HelloReply>(callContext);
+
+            // Act
+            using var call = service.SayHelloServerStreaming(new HelloRequest { Name = "Joe" }, responseStream, callContext);
+
+            // Assert
+            Assert.False(call.IsCompletedSuccessfully, "Method should run until cancelled.");
+
+            cts.Cancel();
+
+            await call;
+            responseStream.Complete();
+
+            var allMessages = new List<HelloReply>();
+            await foreach (var message in responseStream.ReadAllAsync())
+            {
+                allMessages.Add(message);
+            }
+
+            Assert.True(allMessages.Count >= 1);
+
+            Assert.Equal("Hello Joe 1", allMessages[0].Message);
+        }
+
+        [Fact]
+        public async Task SayHelloClientStreamingTest()
+        {
+            // Arrange
+            var mockGreeter = new Mock<IGreeter>();
+            mockGreeter.Setup(m => m.Greet(It.IsAny<string>())).Returns((string s) => $"Hello {s}");
+            var service = new TesterService(mockGreeter.Object);
+
+            var callContext = TestServerCallContext.Create();
+            var requestStream = new TestAsyncStreamReader<HelloRequest>(callContext);
+
+            // Act
+            using var call = service.SayHelloClientStreaming(requestStream, callContext);
+
+            requestStream.AddMessage(new HelloRequest { Name = "James" });
+            requestStream.AddMessage(new HelloRequest { Name = "Jo" });
+            requestStream.AddMessage(new HelloRequest { Name = "Lee" });
+            requestStream.Complete();
+
+            // Assert
+            var response = await call;
+            Assert.Equal("Hello James, Jo, Lee", response.Message);
+        }
+
+        [Fact]
+        public async Task SayHelloBidirectionStreamingTest()
+        {
+            // Arrange
+            var mockGreeter = new Mock<IGreeter>();
+            mockGreeter.Setup(m => m.Greet(It.IsAny<string>())).Returns((string s) => $"Hello {s}");
+            var service = new TesterService(mockGreeter.Object);
+
+            var callContext = TestServerCallContext.Create();
+            var requestStream = new TestAsyncStreamReader<HelloRequest>(callContext);
+            var responseStream = new TestServerStreamWriter<HelloReply>(callContext);
+
+            // Act
+            using var call = service.SayHelloBidirectionalStreaming(requestStream, responseStream, callContext);
```

Comment:
```suggestion
            using var call = service.SayHelloBidirectionalStreaming(requestStream, responseStream, 
                callContext);
```

Reply:
@JamesNK , just got back from a long vacation.  Catching up here.  Guardrex has great review feedback.  Taking a look too to see if I can find anything else just in case and then moving onto next doc.